### PR TITLE
update mergify rules

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

as discussed with @awaelchli 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [ ] Did you list all the breaking changes introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -36,6 +36,26 @@ pull_request_rules:
...
+      - -conflict
+      - -draft # filter-out GH draft PRs
+      - -title~=(?i)wip # skip all PR that title contains “WIP” (ignoring case)
+      - "#approved-reviews-by>=1" # number of review approvals
+      - "#changes-requested-reviews-by==0" # no requested changes
+    actions:
+      label:
+        add: [ "0:] Ready-To-Go" ]
+
+  - name: Need some work
```
Comment (carmocca) [inline .github/mergify.yml:50]:
```suggestion
  - name: Not ready yet
```

File: .github/mergify.yml [hunk 2]
```diff
@@ -36,6 +36,26 @@ pull_request_rules:
...
+      - or:
+        - "#approved-reviews-by==0" # number of review approvals
+        - "#changes-requested-reviews-by>=1" # no requested changes
+    actions:
+      label:
+        remove: [ "0:] Ready-To-Go" ]
+
   #- name: update PR
   #  conditions:
   #    - -conflict
```
Comment (carmocca) [inline .github/mergify.yml:61]: Remove this commented out lines?

**Review by awaelchli — APPROVED** [review]
:)

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
Shouldn't the bot label this PR as ready-to-go now? Or does it have to be merged first?

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
> Shouldn't the bot label this PR as ready-to-go now? Or does it have to be merged first?

it has to be on master first, but you can test it in here - https://dashboard.mergify.io/github/PyTorchLightning/config-editor?repositoryName=pytorch-lightning&pullRequestNumber=8443

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**